### PR TITLE
planner: Move the Selectivity function from the stats package into cardinality package

### PR DESCRIPTION
**Pull request by qw4990**
### What problem does this PR solve?

Issue Number: ref #46358

Problem Summary: Move the Selectivity function from the stats package into cardinality package

### What is changed and how it works?

**No logical change, just some code movement.**

Follow with https://github.com/pingcap/tidb/pull/46359

Move the Selectivity function from the stats package.

image

### Check List

Tests 

- [ ] Unit test
- [x] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: planner/cardinality/histogram.go [hunk 1]
```diff
@@ -0,0 +1,194 @@
+package cardinality
```
Comment (hawkingrei) [inline planner/cardinality/histogram.go:1]: add license header.

File: planner/cardinality/histogram.go [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+				continue
+			}
+			newColl.Indices[node.ID] = newIdxHist
+			continue
+		}
+		oldCol, ok := coll.Columns[node.ID]
+		if !ok {
+			continue
+		}
+		newCol := &statistics.Column{
```
Comment (hawkingrei) [inline planner/cardinality/histogram.go:49]: We can add NewColumn for it.

Comment (qw4990) [reply planner/cardinality/histogram.go:49]: That's exactly what I want to do later on... Use a standard NewFunc to create these structures.

File: planner/cardinality/histogram.go (hunk 1 above)
Comment (hawkingrei) [reply planner/cardinality/histogram.go:1]: remember to run the ```make bazel_prepare```.

**Review by hi-rustin — APPROVED** [review]

**Review by hawkingrei — APPROVED** [review]

**Pull request merged**